packages/kbn-securitysolution-utils/src/path_validations/index.ts

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { i18n } from '@kbn/i18n';
+
+export const FILENAME_WILDCARD_WARNING = i18n.translate('utils.filename.wildcardWarning', {
+  defaultMessage: `A wildcard in the filename will affect the endpoint's performance`,
+});
+
+export const FILEPATH_WARNING = i18n.translate('utils.filename.pathWarning', {
+  defaultMessage: `Path may be formed incorrectly; verify value`,
+});
+
+export enum ConditionEntryField {
+  HASH = 'process.hash.*',
+  PATH = 'process.executable.caseless',
+  SIGNER = 'process.Ext.code_signature',
+}
+
+export enum OperatingSystem {
+  LINUX = 'linux',
+  MAC = 'macos',
+  WINDOWS = 'windows',
+}
+
+type TrustedAppEntryTypes = 'match' | 'wildcard';
+/*
+ * regex to match executable names
+ * starts matching from the eol of the path
+ * file names with a single or multiple spaces (for spaced names)
+ * and hyphens and combinations of these that produce complex names
+ * such as:
+ * c:\home\lib\dmp.dmp
+ * c:\home\lib\my-binary-app-+/ some/  x/ dmp.dmp
+ * /home/lib/dmp.dmp
+ * /home/lib/my-binary-app+-\ some\  x\ dmp.dmp
+ */
+export const WIN_EXEC_PATH = /\\(\w+|\w*[\w+|-]+\/ +)+\w+[\w+|-]+\.*\w+$/i;
```

Comment:
See below but this one is prone to a [ReDOS attack](https://owasp.org/www-community/attacks/Regular_expression_Denial_of_Service_-_ReDoS) as well:
<img width="830" alt="Screen Shot 2022-02-28 at 12 30 46 PM" src="https://user-images.githubusercontent.com/1151048/156046849-4fab35da-e8f5-4946-875e-bc3e2156c058.png">

In this example I ran it against an automated checker:
https://devina.io/redos-checker

I then confirmed this by running it within NodeJS:
```sh
$ node
> var WIN_EXEC_PATH = /\\(\w+|\w*[\w+|-]+\/ +)+\w+[\w+|-]+\.*\w+$/i;
> WIN_EXEC_PATH.test('\\0' + '0'.repeat(31) + '\\00+0');
```

Reply:
fixed f1cc94cc7482aed43d7be4218f5d4295215d1367